db/schema_tables.cc

```diff
@@ -1637,25 +1670,61 @@ static shared_ptr<cql3::functions::user_aggregate> create_aggregate(replica::dat
     if (initcond_str) {
         initcond = state_type->from_string(initcond_str.value());
     }
+    ::shared_ptr<cql3::functions::scalar_function> reduce_func = nullptr;
+    if (rs && !rs->empty()) {
+        auto&& rrow = rs->row(0);
+        auto rfunc = rrow.get_nonnull<sstring>("reduce_func");
 
-    return ::make_shared<cql3::functions::user_aggregate>(name, initcond, std::move(state_func), nullptr, std::move(final_func));
+        reduce_func = dynamic_pointer_cast<cql3::functions::scalar_function>(
+            cql3::functions::functions::find(cql3::functions::function_name{name.keyspace, rfunc}, {state_type, state_type}));
 
+        if (!reduce_func) {
+            throw std::runtime_error(format("Reduce function {} needed by aggregate {} not found", rfunc, name.name));
```

Comment:
Throwing here is risky. It can end up preventing a node from booting.

Maybe it's better just not to create the aggregate (but log an error instead).

Reply:
Because of before/after schema, reduce function is not adding during creation of UDA